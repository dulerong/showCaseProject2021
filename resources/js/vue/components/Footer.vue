<template lang="pug">
v-footer(padless absolute app)
  v-card(width="100%")
    v-card-text
      v-row(no-gutters justify="center")
        v-btn(v-for="(item, i) in $_footer_navigation" :key="i" icon class="mx-4" :to="{ name: item.name, params: { lang: $i18n.locale } }" exact)
          ToolTip(:message="$t(`footer.tooltip.${item.name}`)")
            v-icon {{ item.icon }}
    v-divider
    v-card-text
      v-row(no-gutters justify="center")
        h1 {{ $_dateToday.format('YYYY') }} - Codey Du
</template>

<script>
export default {
  components: {
    ToolTip: require('~components/ToolTip').default,
  },
}
</script>