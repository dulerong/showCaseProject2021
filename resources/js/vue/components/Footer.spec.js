import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'

import Footer from '~components/Footer'

describe('Footer Component', () => {
  const localVue = createLocalVue()
  const footerNavigation = [1, 2, 3]

  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const mountFunction = options => {
    return mount(Footer, {
      localVue,
      vuetify,
      mocks: {
        $t: msg => msg,
        $_dateToday: {
          format: date => date
        },
        $i18n: {
          locale: 'en'
        },
        $_footer_navigation: footerNavigation
      },
      stubs: ['router-link'],
      ...options
    })
  }

  it('Renders properly', () => {
    const wrapper = mountFunction()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Number of footer route is correct', () => {
    const wrapper = mountFunction()

    const buttons = wrapper.findAll('.v-btn')
    expect(buttons.length).toEqual(footerNavigation.length)
  })
})