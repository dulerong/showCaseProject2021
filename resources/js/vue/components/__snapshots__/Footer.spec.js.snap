// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Renders properly 1`] = `
<footer class="v-footer v-sheet v-sheet--outlined theme--light v-footer--absolute v-footer--padless" style="left: 0px; right: 0px; bottom: 0px;">
  <div class="v-card v-sheet theme--light" style="width: 100%;">
    <div class="v-card__text">
      <div class="row no-gutters justify-center">
        <router-link-stub class="mx-4 v-btn v-btn--icon v-btn--round v-btn--router theme--light v-size--default"><span class="v-btn__content"><span class="v-tooltip v-tooltip--bottom"><!----><div aria-haspopup="true" aria-expanded="false"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light"></i></div></span></span></router-link-stub>
        <router-link-stub class="mx-4 v-btn v-btn--icon v-btn--round v-btn--router theme--light v-size--default"><span class="v-btn__content"><span class="v-tooltip v-tooltip--bottom"><!----><div aria-haspopup="true" aria-expanded="false"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light"></i></div></span></span></router-link-stub>
        <router-link-stub class="mx-4 v-btn v-btn--icon v-btn--round v-btn--router theme--light v-size--default"><span class="v-btn__content"><span class="v-tooltip v-tooltip--bottom"><!----><div aria-haspopup="true" aria-expanded="false"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light"></i></div></span></span></router-link-stub>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-card__text">
      <div class="row no-gutters justify-center">
        <h1>YYYY - Codey Du</h1>
      </div>
    </div>
  </div>
</footer>
`;
